Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp

```diff
@@ -454,13 +449,13 @@ namespace AzToolsFramework::ViewportUi::Internal
         // set viewport ui visibility depending on if elements are present
         if (region.isEmpty() || !UiDisplayEnabled())
         {
-            m_uiMainWindow.setVisible(false);
-            m_uiOverlay.setVisible(false);
+            m_uiMainWindow.hide();
+            m_uiOverlay.hide();
         }
         else
         {
-            m_uiMainWindow.setVisible(true);
-            m_uiOverlay.setVisible(true);
+            m_uiMainWindow.show();
+            m_uiOverlay.show();
         }
```

Comment:
`show()` and `hide()` internally just call `setVisible(true)` and `setVisible(false)`. So you can take out the repeated code in the branches of the conditional with this:

```cpp
const bool shouldShow = !region.isEmpty() && UiDisplayEnabled();
m_uiMainWindow.setVisible(shouldShow);
m_uiOverlay.setVisible(shouldShow);
```

Reply:
Yes that's a nice idea, will update 👍